src/extension/app/utils/helix-admin.js

```diff
@@ -10,21 +10,130 @@
  * governing permissions and limitations under the License.
  */
 
+/**
+ * @typedef {import('../store/app.js').AppStore} AppStore
+ */
+
 /**
  * @typedef {import('../store/site.js').SiteStore} SiteStore
  */
 
 /**
- * Creates an Admin URL for an API and path.
+ * Enumeration of rate limiting sources.
  * @private
+ * @type {Object<string, string>}}
+ */
+const RATE_LIMITER = {
+  NONE: '',
+  ADMIN: 'AEM',
+  ONEDRIVE: 'Microsoft SharePoint',
+};
+
+/**
+ * Returns an error message from the server response.
+ * @param {Response} resp The response
+ * @returns {string} The error message or an empty string
+ */
+export function getServerError(resp) {
+  return resp?.headers?.get('x-error') || '';
+}
+
+/**
+ * Checks if the request has been rate-limited and returns the source.
+ * @param {Response} resp The response
+ * @returns {string} The rate limiter (see {@link RATE_LIMITER})
+ */
+function getRateLimiter(resp) {
```

Comment:
I think anything that is not exported we should move into the class. Better for encapsulation, mocking/testing and if we need to it's easier to access class properties.